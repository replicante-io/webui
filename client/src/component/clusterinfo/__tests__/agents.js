'use strict';
//@flow

import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Agents from '../agents';


const mockStore = configureStore([]);
const MATCH = {
  params: {name: 'test'}
};


describe('ClusterInfo', () => {
  describe('agents', () => {
    let store;

    beforeEach(() => {
      store = mockStore({
        clusterinfo: {
          agents: {}
        }
      });
    });

    test('renders no data', () => {
      const tree = renderer.create(
        <Provider store={store}>
          <Agents match={MATCH} />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders with data', () => {
      store = mockStore({ clusterinfo: { agents: {
        'test': [{
          host: 'http://host1:1234',
          status: { code: 'UP' },
          version_checkout: null,
          version_number: 'abc',
          version_taint: null,
        }]
      } } });
      const tree = renderer.create(
        <Provider store={store}>
          <Agents match={MATCH} />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
