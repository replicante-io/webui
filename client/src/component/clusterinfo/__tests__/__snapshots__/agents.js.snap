// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterInfo agents renders no data 1`] = `
<div
  className="text-center"
>
  <p
    className="large-icon nodata"
  >
    <span
      className="fa-layers fa-fw"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-database fa-w-14 "
        data-icon="database"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 73.143v45.714C448 159.143 347.667 192 224 192S0 159.143 0 118.857V73.143C0 32.857 100.333 0 224 0s224 32.857 224 73.143zM448 176v102.857C448 319.143 347.667 352 224 352S0 319.143 0 278.857V176c48.125 33.143 136.208 48.572 224 48.572S399.874 209.143 448 176zm0 160v102.857C448 479.143 347.667 512 224 512S0 479.143 0 438.857V336c48.125 33.143 136.208 48.572 224 48.572S399.874 369.143 448 336z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-question fa-w-12 overlay"
        data-icon="question"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </p>
  No information available. 
  <br />
  If nothing shows up the cluster may not exist.
</div>
`;

exports[`ClusterInfo agents renders with data 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Address
      </th>
      <th>
        Status
      </th>
      <th>
        Version
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        http://host1:1234
      </td>
      <td>
        UP
        <br />
      </td>
      <td>
        <div>
          SemVer: 
          abc
        </div>
      </td>
    </tr>
    <tr>
      <td>
        http://host1:1234
      </td>
      <td>
        DOWN
        <br />
        <small>
          some reason
        </small>
      </td>
      <td>
        <div>
          Git SHA: 
          def
        </div>
        <div>
          Git taint indicator: 
          yes
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
